core/src/main/scala/org/http4s/rho/ServiceBuilder.scala

```diff
@@ -7,15 +7,15 @@ import shapeless.HList
 import scala.collection.immutable.VectorBuilder
 
 /** CompileService which accumulates routes and can build a `HttpService` */
-final class ServiceBuilder[F[_]: Monad] private(internalRoutes: VectorBuilder[RhoRoute.Tpe[F]]) extends CompileService[F, RhoRoute.Tpe[F]] {
+final class ServiceBuilder[F[_]: Monad] private(internalRoutes: VectorBuilder[RhoRoute.Tpe[F]]) extends CompileRoutes[F, RhoRoute.Tpe[F]] {
 
   /** Turn the accumulated routes into an `HttpService`
     *
     * @param filter [[RhoMiddleware]] to apply to the collection of routes.
     * @return An `HttpService` which can be mounted by http4s servers.
     */
   def toService(filter: RhoMiddleware[F] = identity): HttpRoutes[F] =
```

Comment:
`ServiceBuilder` and its `toService` function also need a rename.

Reply:
Agree, but I think we should handle this along with `RhoService`

Reply:
Yeah, I agree with the `RhoService` rename.  It makes sense in context with what http4s is trying to do.

Reply:
My worry is similar: we're getting too many "route" in different places that clash with each other and core's. We're ending up with `RouteFoo.toRoutes(routes).asRoutes` roughly speaking. But still I believe this is a right choice.